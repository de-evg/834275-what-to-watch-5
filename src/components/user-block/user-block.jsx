import React from "react";
import {AuthorizationStatus} from "../../const";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import userBlockProps from "./user-block.props";

const UserBlock = ({authorizationStatus, userAvatar}) => {
  return (
    <div className="user-block">
      {
        authorizationStatus === AuthorizationStatus.AUTH
          ? (
            <div className="user-block__avatar">
              <Link to={`/myList`} >
                <img src={userAvatar} alt="User avatar" width="63" height="63" />
              </Link>
            </div>
          )
          : <Link to="/login" className="user-block__link">Sign in</Link>
      }
    </div>
  );
};

UserBlock.propTypes = userBlockProps;

const mapStateToProps = (state) => ({
  authorizationStatus: state.USER.authorizationStatus,
  userAvatar: state.USER.userAvatar,
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
