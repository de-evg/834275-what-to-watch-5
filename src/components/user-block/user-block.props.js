import PropTypes from "prop-types";

const userBlockProps = {
  authorizationStatus: PropTypes.string.isRequired,
  userAvatar: PropTypes.string.isRequired,
};

export default userBlockProps;
