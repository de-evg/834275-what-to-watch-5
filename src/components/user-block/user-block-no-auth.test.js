import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {UserBlock} from "./user-block";

it(`Should UserBlock render correctly on unauthorisation`, () => {
  const tree = renderer
    .create(
        <BrowserRouter><UserBlock
          authorizationStatus={`NO_AUTH`}
          userAvatar={`img/avatar.jpg`}
        />
        </BrowserRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
