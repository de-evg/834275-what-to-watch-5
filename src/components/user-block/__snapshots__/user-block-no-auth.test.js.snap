// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly on unauthorisation 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
